protocol/protocol.go

```diff
@@ -29,7 +29,7 @@ import (
 // Extension - protocol
 type Protocol interface {
 	Export(invoker Invoker) Exporter
-	Refer(url common.URL) Invoker
+	Refer(url common.URL, impl interface{}) Invoker
 	Destroy()
```

Comment:
Is there any other solution rather than add a parameter? Too many code changed.. 
And it's also not commonly used except for grpc.